src/language-js/needs-parens.js

```diff
@@ -711,7 +711,8 @@ function needsParens(path, options) {
         (parent.type === "BindExpression" &&
           name === "callee" &&
           parent.callee === node) ||
-        parent.type === "MemberExpression"
+        parent.type === "MemberExpression" ||
+        parent.type === "NewExpression"
```

Comment:
You also need to check that it's the callee, so that `new f(a::b)` doesn't end up as `new f((a::b))`. Should probably add a test for that case, too.